packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx

```diff
@@ -731,6 +731,18 @@ function CodeMirror(props: NoteBodyEditorProps, ref: any) {
 	// It might be buggy, refer to the below issue
 	// https://github.com/laurent22/joplin/pull/3974#issuecomment-718936703
 	useEffect(() => {
+		let touched = false;
+
+		function touchedEditor(event: any) {
+
+			if (event.target.closest('.codeMirrorEditor') !== null) {
+				touched = true;
+			} else {
+				touched = false;
+			}
+		}
+		document.addEventListener('contextmenu', touchedEditor);
```

Comment:
Doesn't it mean that we should use this contextmenu event, instead of the context-menu event we're currently using? It seems strange to have to support both.

Reply:
I found another solution by using the "params" of "onContextMenu" function and checking if "isEditable" is true that means can write and when the 'inputFieldType' is equal to plaintext or password that means there exist a layout like NOTEBOOKS +, change the app layout, or enter Hyperlink  above the codeMirrorEditor